File: frontend/cypress/test-solutions/find-all-anagrams-in-a-string.py
class Solution:
    def findAnagrams(self, s: str, p: str) -> list[int]:
        if len(p) > len(s):
            return []
        
        result = []
        p_count = {}
        window_count = {}
        
        for char in p:
            p_count[char] = p_count.get(char, 0) + 1
        
        window_size = len(p)
        
        for i in range(len(s)):
            window_count[s[i]] = window_count.get(s[i], 0) + 1
            
            if i >= window_size:
                left_char = s[i - window_size]
                window_count[left_char] -= 1
                if window_count[left_char] == 0:
                    del window_count[left_char]
            
            if window_count == p_count:
                result.append(i - window_size + 1)
        
        return result

File: backend/test-solutions/find-all-anagrams-in-a-string.js
class Solution {
    findAnagrams(s, p) {
        if (p.length > s.length) return [];
        
        const result = [];
        const pCount = new Map();
        const windowCount = new Map();
        
        for (const char of p) {
            pCount.set(char, (pCount.get(char) || 0) + 1);
        }
        
        const windowSize = p.length;
        
        for (let i = 0; i < s.length; i++) {
            windowCount.set(s[i], (windowCount.get(s[i]) || 0) + 1);
            
            if (i >= windowSize) {
                const leftChar = s[i - windowSize];
                windowCount.set(leftChar, windowCount.get(leftChar) - 1);
                if (windowCount.get(leftChar) === 0) {
                    windowCount.delete(leftChar);
                }
            }
            
            if (this.mapsEqual(windowCount, pCount)) {
                result.push(i - windowSize + 1);
            }
        }
        
        return result;
    }
    
    mapsEqual(map1, map2) {
        if (map1.size !== map2.size) return false;
        for (const [key, value] of map1) {
            if (map2.get(key) !== value) return false;
        }
        return true;
    }
}